teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,473 @@
+---
+title: Tekton Enhancement Proposal Process
+authors:
+  - "@vdemeester"
+creation-date: 2020-03-10
+last-updated: 2020-04-23
+status: proposed
+---
+
+# Tekton Enhancement Proposal Process
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Stewardship](#stewardship)
+- [Reference-level explanation](#reference-level-explanation)
+  - [What type of work should be tracked by a TEP](#what-type-of-work-should-be-tracked-by-a-tep)
+  - [TEP Template](#tep-template)
+  - [TEP Metadata](#tep-metadata)
+  - [TEP Workflow](#tep-workflow)
+  - [Git Implementation](#git-implementation)
+  - [Prior Art](#prior-art)
+- [Examples](#examples)
+  - [Share Task and Pipeline as OCI artifact](#share-task-and-pipeline-as-oci-artifact)
+  - [PipelineResource re-design](#pipelineresource-re-design)
+- [Drawbacks](#drawbacks)
+- [Unresolved Questions](#unresolved-questions)
+<!-- /toc -->
+
+
+## Summary
+
+A standardized development process for Tekton is proposed in order to
+
+- provide a common structure and clear checkpoints for proposing
+  changes to Tekton
+- ensure that the motivation for a change is clear
+- allow for the enumeration stability milestones and stability
+  graduation criteria
+- persist project information in a Version Control System (VCS) for
+  future Tekton users and contributors
+- support the creation of _high value user facing_ information such
+  as:
+  - an overall project development roadmap
+  - motivation for impactful user facing changes
+- ensure community participants are successfully able to drive changes
+  to completion across one or more releases while stakeholders are
+  adequately represented throughout the process
+
+This process is supported by a unit of work called a Tekton
+Enhancement Proposal (TEP). A TEP attempts to combine aspects of the
+following:
+
+- feature, and effort tracking document
+- a product requirements document
+- design document
```

Comment:
seems kinda weird that this says the TEP should contain the design doc and requirements, while below we say you may still need design docs (and i think design docs should contain information that the TEP may not, e.g. analysis of alternatives)

i feel like you could remove these bullet points without losing much info tho, maybe that's clearer?

Reply:
:thinking: It doesn't say it "should contain", it says "it combines" as you can see a TEP as "…" (the following bullet point)

> and i think design docs should contain information that the TEP may not, e.g. analysis of alternatives

Why would a TEP not include alternatives ? (or link to deeper analysis doc)

Reply:
just ignore me, i meant to remove that part of the comment and forgot to get back to it - before i saw the template i didnt realize how much detail we were putting into the TEP and i definitely like including alternatives in the TEP, i think that's great! sorry for the confusion